pkg/vault/test/foundry/utils/BaseERC4626BufferTest.sol

```diff
@@ -118,4 +192,30 @@ abstract contract BaseERC4626BufferTest is BaseVaultTest {
         IERC20(address(erc4626Pool)).approve(address(batchRouter), type(uint256).max);
         vm.stopPrank();
     }
+
+    function _setupWrappedTokens() private {
+        waDAI = new ERC4626TestToken(dai, "Wrapped aDAI", "waDAI", 18);
+        vm.label(address(waDAI), "waDAI");
+
+        // "USDC" is deliberately 18 decimals to test one thing at a time.
+        waUSDC = new ERC4626TestToken(usdc, "Wrapped aUSDC", "waUSDC", 18);
+        vm.label(address(waUSDC), "waUSDC");
+
+        (waDaiIdx, waUsdcIdx) = getSortedIndexes(address(waDAI), address(waUSDC));
+
+        // Manipulate rates before creating the ERC4626 pools. It's important to not have a 1:1 rate when testing
+        // ERC4626 tokens, so we can differentiate between wrapped and underlying amounts.
+        dai.mint(lp, 10 * erc4626PoolInitialAmount);
+        usdc.mint(lp, 10 * erc4626PoolInitialAmount);
+        // Deposit sets the rate to 1.
+        vm.startPrank(lp);
+        dai.approve(address(waDAI), 10 * erc4626PoolInitialAmount);
+        waDAI.deposit(10 * erc4626PoolInitialAmount, lp);
+        usdc.approve(address(waUSDC), 10 * erc4626PoolInitialAmount);
+        waUSDC.deposit(10 * erc4626PoolInitialAmount, lp);
+        vm.stopPrank();
+        // Changing asset balances without minting shares changes the rate so that it is no longer 1.
+        dai.mint(address(waDAI), 2 * erc4626PoolInitialAmount);
+        usdc.mint(address(waUSDC), 4 * erc4626PoolInitialAmount);
```

Comment:
I know we're already converting back and forth between assets and shares, but can we add a very explicit assertion that checks that the rate of the wrappers is not FP(1)?